assets/src/edit-story/components/library/panes/pageLayouts/pageLayouts.js

```diff
@@ -42,33 +47,42 @@ const PageLayoutsContainer = styled.div`
   position: relative;
 `;
 
-const PageLayoutsRow = styled.div`
+const PageLayoutsVirtualizedContainer = styled.div`
   position: absolute;
   top: 0;
   left: ${PANE_PADDING};
-  height: ${({ height }) => `${height}px`};
-  transform: ${({ translateY }) => `translateY(${translateY}px)`};
   display: grid;
   grid-template-columns: 1fr 1fr;
-  column-gap: 12px;
+  grid-template-columns: ${({ pageSize }) => `
+    repeat(auto-fill, ${pageSize.width}px)`};
+  grid-template-rows: ${({ pageSize }) =>
+    `minmax(${pageSize.containerHeight}px, auto)`};
+  gap: ${PAGE_LAYOUT_ROW_GAP}px;
+  width: 100%;
+  height: 100%;
 `;
 
-function PageLayouts(props) {
-  const { pages, parentRef } = props;
-
+function PageLayouts({ pages, parentRef }) {
   const { replaceCurrentPage, currentPage } = useStory(
     ({ actions: { replaceCurrentPage }, state: { currentPage } }) => ({
       replaceCurrentPage,
       currentPage,
     })
   );
 
-  const handleApplyPageLayout = useCallback(
-    (page) => {
-      const duplicatedPage = duplicatePage(page);
-      replaceCurrentPage({ page: duplicatedPage });
-    },
-    [replaceCurrentPage]
+  const containerRef = useRef();
+  const pageRefs = useRef({});
+
+  const [activePage, setActivePage] = useState();
```

Comment:
This is confusing that `activePage` is the one that was clicked on and `activePageId` is the one with current focus. Can you distinguish these names maybe rename `activePage` to `selectedPage`?

I think that having `activePage` and `isConfirming` are redundant since it seems they are both set and unset together. If that's the case then I would advocate for just tracking a `confirmingPage`.